From eeea34eeaba03489d8303ed58bc3a411686e2c98 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Fri, 18 Jan 2013 10:14:25 +0800
Subject: [PATCH 02/13] imx-pcm-dma: DMA channel parameter depend on the snd_soc_dai

Since both S/PDIF and SGTL5000 use the same snd_pcm_ops operation to
transfer data. But the dma channel parameter depend on the snd_soc_dai.

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 sound/soc/imx/imx-pcm-dma-mx2.c |    3 +--
 sound/soc/imx/imx-pcm.h         |    3 ++-
 sound/soc/imx/imx-ssi.c         |    2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/imx/imx-pcm-dma-mx2.c b/sound/soc/imx/imx-pcm-dma-mx2.c
index f3c0a5e..0343e99 100644
--- a/sound/soc/imx/imx-pcm-dma-mx2.c
+++ b/sound/soc/imx/imx-pcm-dma-mx2.c
@@ -109,8 +109,7 @@ static int snd_imx_open(struct snd_pcm_substream *substream)
 	dma_params = snd_soc_dai_get_dma_data(rtd->cpu_dai, substream);
 
 	dma_data = kzalloc(sizeof(*dma_data), GFP_KERNEL);
-	dma_data->peripheral_type = dma_params->shared_peripheral ?
-					IMX_DMATYPE_SSI_SP : IMX_DMATYPE_SSI;
+	dma_data->peripheral_type = dma_params->shared_peripheral;
 	dma_data->priority = DMA_PRIO_HIGH;
 	dma_data->dma_request = dma_params->dma;
 
diff --git a/sound/soc/imx/imx-pcm.h b/sound/soc/imx/imx-pcm.h
index 79b0f5c..eb72ced 100644
--- a/sound/soc/imx/imx-pcm.h
+++ b/sound/soc/imx/imx-pcm.h
@@ -13,6 +13,7 @@
 #ifndef _IMX_PCM_H
 #define _IMX_PCM_H
 
+#include <mach/dma.h>
 /*
  * Do not change this as the FIQ handler depends on this size
  */
@@ -22,7 +23,7 @@ struct imx_pcm_dma_params {
 	int dma;
 	unsigned long dma_addr;
 	int burstsize;
-	bool shared_peripheral; /* The peripheral is on SPBA bus */
+	enum sdma_peripheral_type shared_peripheral; /* The peripheral is on SPBA bus */
 };
 
 int snd_imx_pcm_mmap(struct snd_pcm_substream *substream,
diff --git a/sound/soc/imx/imx-ssi.c b/sound/soc/imx/imx-ssi.c
index befe902..2c941c3 100644
--- a/sound/soc/imx/imx-ssi.c
+++ b/sound/soc/imx/imx-ssi.c
@@ -586,7 +586,7 @@ static int imx_ssi_probe(struct platform_device *pdev)
 	ssi->dma_params_rx.dma_addr = res->start + SSI_SRX0;
 	ssi->dma_params_tx.dma_addr = res->start + SSI_STX0;
 
-	ssi->dma_params_tx.shared_peripheral = true;
+	ssi->dma_params_tx.shared_peripheral = IMX_DMATYPE_SSI_SP;
 	ssi->dma_params_rx.shared_peripheral =
 			ssi->dma_params_tx.shared_peripheral;
 
-- 
1.7.0

